rust-mode.el

```diff
@@ -1332,29 +1332,49 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
 ;; Formatting using rustfmt
 (defun rust--format-call (buf)
   "Format BUF using rustfmt."
-  (with-current-buffer (get-buffer-create "*rustfmt*")
-    (erase-buffer)
-    (insert-buffer-substring buf)
-    (let* ((tmpf (make-temp-file "rustfmt"))
```

Comment:
Do you know why this code previously used a temporary file? I wonder if there was some issue with using stdin.

Reply:
Maybe this rust-lang-nursery/rustfmt#562 ?

Reply:
Ok this can't be it, as the old version also uses stdin. Maybe the author preferred `call-process-region` or didn't know about `shell-command-on-region`.